src/tree/hist/histogram.h

```diff
@@ -76,6 +84,20 @@ class HistogramBuilder {
       buffer_.Reset(this->n_threads_, n_nodes, space, target_hists);
     }
 
+    constexpr float kColsampleTh = 0.1;
+    bool column_sampling = (column_sampler_ != nullptr) &&
+                           (train_param_.colsample_bytree < kColsampleTh ||
+                            train_param_.colsample_bylevel < kColsampleTh);
```

Comment:
how about bynode? Is it used?

Reply:
Not now, maybe later, one can investigate these options more deeply.